core/io/http_session.hxx

```diff
@@ -349,13 +349,14 @@ class http_session : public std::enable_shared_from_this<http_session>
             if (ec == asio::error::operation_aborted) {
                 return;
             }
+            CB_LOG_DEBUG("{} idle timeout expired, stopping session: \"{}:{}\"", self->info_.log_prefix(), self->hostname_, self->service_);
             self->stop();
         });
     }
 
-    void reset_idle()
+    std::size_t reset_idle()
```

Comment:
Since http_session internally manages the idle_timer, you should probably do the conversion from 'arbitrary number' to boolean here, where the context for that number is known.

Reply:
Fair -- I will make that change.